templates/examples/extensions/py_lambda_hooks/CanvasHook/CanvasHook.py

```diff
@@ -0,0 +1,405 @@
+from canvasapi import Canvas
+
+
+MATCHING_TOLERANCE_SCORE = 70   #used for matching accuracy with fuzzy match
+
+#----------------------------------------------------------------------
+# function: get_secret
+#----------------------------------------------------------------------
+def get_secret(secrets_id_name):
+
+    region_name = os.environ['AWS_REGION']
+
+    # Create a Secrets Manager client
+    session = boto3.session.Session()
+    client = session.client(
+        service_name='secretsmanager',
+        region_name=region_name
+    )
+
+    # In this sample we only handle the specific exceptions for the 'GetSecretValue' API.
+    # See https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_GetSecretValue.html
+    # We rethrow the exception by default.
+
+    try:
+        get_secret_value_response = client.get_secret_value(
+            SecretId=secrets_id_name
+        )
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'DecryptionFailureException':
+            # Secrets Manager can't decrypt the protected secret text using the provided KMS key.
+            # Deal with the exception here, and/or rethrow at your discretion.
+            raise e
+        elif e.response['Error']['Code'] == 'InternalServiceErrorException':
+            # An error occurred on the server side.
+            # Deal with the exception here, and/or rethrow at your discretion.
+            raise e
+        elif e.response['Error']['Code'] == 'InvalidParameterException':
+            # You provided an invalid value for a parameter.
+            # Deal with the exception here, and/or rethrow at your discretion.
+            raise e
+        elif e.response['Error']['Code'] == 'InvalidRequestException':
+            # You provided a parameter value that is not valid for the current state of the resource.
+            # Deal with the exception here, and/or rethrow at your discretion.
+            raise e
+        elif e.response['Error']['Code'] == 'ResourceNotFoundException':
+            # We can't find the resource that you asked for.
+            # Deal with the exception here, and/or rethrow at your discretion.
+            raise e
+    else:
+        # Decrypts secret using the associated KMS CMK.
+        # Depending on whether the secret is a string or binary, one of these fields will be populated.
+        if 'SecretString' in get_secret_value_response:
+            secret = get_secret_value_response['SecretString']
+        else:
+            decoded_binary_secret = base64.b64decode(get_secret_value_response['SecretBinary'])
+
+    #return the API token
+    return json.loads(get_secret_value_response['SecretString'])['API_Token']
+
+
+#----------------------------------------------------------------------
+# function to get Canvas User by using  user_id to match with LMS SIS_ID
+#----------------------------------------------------------------------
+
+def getCanvasUser (param_canvas, param_user_name):
```

Comment:
Nice to have: organize all the canvas functions (logic, interface, behavior) into its own Class (or classes). Something like a CanvasWrapper, etc. This would also set good base base for other similar extension.